### ISSUE TITLE
Event API does not update existing proxy entity annotations

### ISSUE THREAD

**Issue by cynici** [sensu/sensu-go#4309]
sensu-backend uses entity annotations provided in event API request only when a proxy entity is first created. It ignores changes in entity annotations supplied in API request when the proxy entity already exists. 

## Expected Behavior

[API documentation](https://docs.sensu.io/sensu-go/latest/api/events/#eventsentitycheck-put) states that
> The /events/:entity/:check API endpoint provides HTTP PUT access to create or update an event and send it to the Sensu pipeline.

Am I correct to presume that sensu-backend is able to merge proxy entity annotations from the latest event API request it receives?

## Current Behavior

I use the `PUT` event API to create a proxy entity with some annotations. Next, I modify the annotations and check status in the JSON payload and made a second `PUT` request. The proxy entity reflects the updated check status but the entity's annotations remain unchanged.

### First PUT payload

``` json
{
  "entity": {
    "entity_class": "proxy",
    "metadata": {
      "name": "test_proxy_entity",
      "annotations": {
        "sensu.io/plugins/sensu-opsgenie-handler/config/priority": "P5",
        "sensu.io/plugins/sensu-opsgenie-handler/config/team": "BlueTeam"
      }
    }
  },
  "check": {
    "handlers": [
      "opsgenie"
    ],
    "interval": 60,
    "executed": 1623152200,
    "issued": 1623152200,
    "output": "some random output",
    "status": 0,
    "metadata": {
      "name": "CHECK-NAME"
    }
  }
}
```

### Second PUT payload

``` json
{
  "entity": {
    "entity_class": "proxy",
    "metadata": {
      "name": "test_proxy_entity",
      "annotations": {
        "sensu.io/plugins/sensu-opsgenie-handler/config/priority": "P1",
        "sensu.io/plugins/sensu-opsgenie-handler/config/team": "RedTeam"
      }
    }
  },
  "check": {
    "handlers": [
      "opsgenie"
    ],
    "interval": 60,
    "executed": 1623152280,
    "issued": 1623152280,
    "output": "some random output",
    "status": 2,
    "metadata": {
      "name": "CHECK-NAME"
    }
  }
}
```

## Context

In my setup, I need to be able to change the Opsgenie team and priority of an event (same entity and check name) based on other conditions upstream. So, I need to be able to update entity annotations of any existing proxy entities by referring to the entity name and check name.

## Your Environment
* Sensu version used (sensuctl, sensu-backend): v6.3.0
* Installation method (packages, binaries, docker etc.): binary compiled from source repo on Github, tagged v6.3.0
* Operating System and version: Ubuntu 20.04

**Comment by cynici** [comment]
Hi @calebhailey 
Have you been able to replicate this issue?
Do you have any advice how a proxy-entity could update its annotations via event API?

**Comment by cynici** [comment]
Answering my own question.

The correct way to specify labels/annotations (which vary over time for the same check) for proxy-entity check via Event API is insert them into the `check.metadata` block, and never in the `entity.metadata` block.

The mistake I'd made was to specify labels/annotations in `entity.metadata` block of proxy entities (created via Event API). In doing so, the labels and annotations never got updated by subsequent check events.

Sorry if I have wasted your time. 🙏
